src/plugins/messenger/MessengerPreview/components/MessengerImage/components/ImagePreview.tsx

```diff
@@ -0,0 +1,64 @@
+import { useMessangerImageContext } from '../MessangerImageContext';
+import { MessagePluginFactoryProps } from "../../../../../../common/interfaces/message-plugin";
+import { getBackgroundImage } from '../../../lib/css';
+
+export const getImagePreview = ({ React, styled }: MessagePluginFactoryProps) => {
+
+    const templateStyle = ({ template = 'media' }) => ({
+        media: {
+            paddingTop: "75%"
+        },
+        list: {
+            paddingTop: "50%"
+        },
+        generic: {
+            paddingTop: "50%",
+            display: "table",
+            width: "100%"
+        }
+    })[template];
+
+    const FixedImage = styled.div({
+        backgroundSize: "cover",
+        backgroundPosition: "center center",
+        cursor: 'pointer'
+    }, templateStyle)
+
+    const FlexImage = styled.img({
+        display: 'block',
+        width: '100%',
+        cursor: 'pointer'
+    })
+
+    const ImagePreview = () => {
+        const { config, url, altText, template, onExpand } = useMessangerImageContext();
+
+        const handleKeyDown = (event: KeyboardEvent) => {
+            event.code === "Enter" && onExpand && onExpand();
+        }
+
+        return config.settings.dynamicImageAspectRatio ? (
+                <FlexImage
+                    src={url}
+                    alt={altText || "Attachment"}
+                    onClick={() => onExpand()}
+                    data-test="image-preview"
+                    tabIndex="0"
+                    onKeyDown={handleKeyDown}
+                />
+            ) : (
+                <FixedImage
+                    style={{ backgroundImage: getBackgroundImage(url) }}
+                    onClick={() => onExpand()}
+                    template={template}
+                    data-test="image-preview"
```

Comment:
Now that we add tabIndex=0, we must add role="button" and an aria-label to this wrapper. Please do not add the button role and aria-label to FlexImage component, since it uses the image element. We can instead wrap the FlexImage in a div that acts as a button. We need to be sure that adding this div does not break anything.